src/repairnator-jenkins-plugin/src/main/java/io/jenkins/plugins/main/RepairnatorPostBuild.java

```diff
@@ -61,6 +59,17 @@
 import jenkins.model.Jenkins;
 
 import java.util.List;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Date;
+
+import java.nio.file.Files;
+import java.nio.file.attribute.FileTime;
+import java.nio.file.attribute.BasicFileAttributes;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.concurrent.TimeUnit;
 
 /* Post build class for post build action*/
 public class RepairnatorPostBuild extends Recorder {
```

Comment:
It's not a responsibility of RepairnatorPostBuild to handle SonarQube (or any specific repair system).

What about using the design pattern strategy?

Reply:
A refactor will be made. :)